feat: sanitize the version arg in the urls data source

obvionaoe opened the pull request:
This pull request fixes #247 

It parses the provided talos_version argument and checks that it can be converted to a Semantic Version, if it can it is converted into a string and prefixed with "v".

frezbo commented on pkg/talos/talos_image_factory_urls_data_source_test.go:
```diff
@@ -234,3 +273,36 @@ data "talos_image_factory_urls" "this" {
+provider "talos" {}
+
+data "talos_image_factory_urls" "this" {
+	talos_version = "v1.7.5"
```
I guess there should be no `v` prefix here

obvionaoe replied:
You are correct, copy and paste error

obvionaoe commented:
@frezbo What would be the best way of fixing the commit compliance issues?

obvionaoe commented:
@frezbo did that work? Sorry for the delay, had a busy weak

frezbo commented:
> @frezbo did that work? Sorry for the delay, had a busy week

yes, thanks, will take it from here

frezbo commented:
I've fixed up the code to use already used semver libraries and cleaned up the tests to be simpler.

frezbo commented:
/m

obvionaoe commented:
> I've fixed up the code to use already used semver libraries and cleaned up the tests to be simpler.

I understand this change, but I used the Masterminds/semver library because it was already a transient dependency and it was currently maintained. blang/semver does not seem to be maintained as it hasn't received any updates in the last 5 years, at least from what I see on GitHub.
